File: CivitasRuby/lib/casilla_calle.rb
# encoding:utf-8
require_relative 'casilla'
module Civitas
  class CasillaCalle < Casilla
    def initialize (titulo)
      super(titulo.nombre)
      @titulo = titulo 
    end
    
    attr_reader :titulo

    def recibe_jugador(iactual,todos)
      #puts "aqui calle"
      #puts super
      if jugador_correcto(iactual, todos)
          informe(iactual,todos)
          jugador = todos[iactual];

          if !@titulo.tiene_propietario()
            jugador.puede_comprar_casilla()
          else
            @titulo.tramitar_alquiler(jugador)
          end
      end
    end


      def to_s
        "Casilla { \n #{@titulo.to_s} \n}"
      end
  end
end

File: CivitasJava/src/Civitas/CasillaCalle.java
package Civitas;

import java.util.ArrayList;

public class CasillaCalle extends Casilla {
    private TituloPropiedad titulo;
    
    CasillaCalle(TituloPropiedad titulo){
        super(titulo.getNombre());
        this.titulo = titulo;
        super.tipo = 1;
    }
    
    TituloPropiedad getTituloPropiedad(){
        return this.titulo;
    }
    
    @Override
    void recibeJugador(int iactual, ArrayList<Jugador> todos) {
            if(super.jugadorCorrecto(iactual,todos)){
            super.informe(iactual,todos);
            Jugador jugador = todos.get(iactual);
            
            if(!this.titulo.tienePropietario()){
                jugador.puedeComprarCasilla();
            } else{
                this.titulo.tramitarAlquiler(jugador);
            }
        }
    }
    
    @Override
    public String toString(){
        return "Casilla {" + "\n Nombre: " + super.getNombre() +"\n" + titulo.toString() + "}\n";
    }
    
    
    
    
}
